auth-providers/oauth2-provider/src/main/java/tech/ydb/auth/OAuth2TokenExchangeProvider.java

```diff
@@ -0,0 +1,320 @@
+package tech.ydb.auth;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.io.UnsupportedEncodingException;
+import java.time.Clock;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.gson.Gson;
+import com.google.gson.annotations.SerializedName;
+import org.apache.http.HttpEntity;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicNameValuePair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tech.ydb.core.Status;
+import tech.ydb.core.StatusCode;
+import tech.ydb.core.UnexpectedResultException;
+import tech.ydb.core.auth.BackgroundIdentity;
+import tech.ydb.core.impl.auth.GrpcAuthRpc;
+
+/**
+ *
+ * @author Aleksandr Gorshenin
+ */
+public class OAuth2TokenExchangeProvider implements AuthRpcProvider<GrpcAuthRpc> {
```

Comment:
We already have implementations in other SDKs and they have similar interface.
C++: https://github.com/ydb-platform/ydb/blob/main/ydb/public/sdk/cpp/client/ydb_types/credentials/oauth2_token_exchange/credentials.h#L35
Python: https://github.com/ydb-platform/ydb-python-sdk/blob/main/ydb/oauth2_token_exchange/token_exchange.py#L98
Go: https://github.com/ydb-platform/ydb-go-sdk/blob/master/credentials/options.go#L28-L141

All interfaces have TokenSource - interface that can be any token (now Fixed or Jwt) that is exchanged onto access token. And there can be two such token in token exchange protocol - subject token and actor token (https://www.rfc-editor.org/rfc/rfc8693)
Plus there is a set of options that have similar names in all SDKs and are supported: both in JWT token source and in token exchange provider params.

Please do the same in Java SDK